### PR TITLE
ci: initial CI configuration

### PR DESCRIPTION
**Pull request by mrc0mmand**

### CODE REVIEW

**Comment by mrc0mmand** [comment]
Oh well, after a trip to the *Workaroundtown* it finally builds the sysext images as well. I guess the next order of business would be to check if the initrd images are bootable.

@keszybz PTAL.

File: docs/fedora.md [hunk 1]
```diff
@@ -156,7 +156,7 @@ Once that's done, we can build a sysext:
 ```bash
 sudo PYTHONPATH=$PWD/../mkosi python -m mkosi --default fedora.mkosi -f \
   --image-version=$KVER-ssh \
-  --base-image=mkosi.output/initrd_$KVER.d \
+  --base-image=mkosi.output/initrd_$KVER \
```
Comment (keszybz) [inline docs/fedora.md:159]: This doesn't seem to be configured anywhere explicitly. I used `.d` locally, but I think I specified it explicitly. If `mkosi` uses a name without `.d` automatically, let's just do that.

File: .github/workflows/build-fedora.sh [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        DEPS)
+            echo "Installing necessary dependencies"
+            dnf -y install dnf-plugins-core kernel-core mkosi python3-pyxattr systemd zstd
+            ;;
+        INITRD)
+            rm -fr mkosi.output
+
+            # Can't use `uname -r`, since we're in a container
+            KVER="$(rpm -q kernel-core --qf "%{version}-%{release}.%{arch}\n" | sed 1q)"
+            [[ -e "kernel-core-$KVER.rpm" ]] || dnf -y download "kernel-core-$KVER"
```
Comment (keszybz) [inline .github/workflows/build-fedora.sh:20]: This shouldn't be necessary, `mkosi.finalize` downloads the image automatically now.

**Review by keszybz — COMMENTED** [review]
Looks great, thanks!

I think the explicit download should not be necessary, but let's merge this like this. We can then remove it if it turns out that the implicit download works.

File: .github/workflows/build-fedora.sh (hunk 2 above)
Comment (mrc0mmand) [reply .github/workflows/build-fedora.sh:20]: Ah, interesting, didn't know about that. I'll try it with the next batch of CI updates.

**Pull request merged**